aptos-move/move-examples/resource_account/sources/resource_account.move

```diff
@@ -0,0 +1,129 @@
+module resource_account::resource_account {
+    use std::signer;
+    use std::string;
+
+    use aptos_framework::account;
+    use aptos_framework::coin::{Self, Coin, MintCapability, BurnCapability};
+    use aptos_framework::resource_account;
+    use aptos_framework::aptos_coin::{AptosCoin};
+
+    struct ModuleData has key {
+        resource_signer_cap: account::SignerCapability,
+    }
+
+    struct CoinCapabilities has key {
+        burn_cap: BurnCapability<WrappedAptosCoin>,
+        mint_cap: MintCapability<WrappedAptosCoin>,
+    }
+
+    struct WrappedAptosCoin {
+        aptos_coin: AptosCoin
+    }
+
+    const EACCOUNT_DOES_NOT_EXIST: u64 = 0;
+    const EINSUFFICIENT_FUND: u64 = 1;
+    const EINVALID_SIGNER: u64 = 2;
+    const EINVALID_COIN_VALUE: u64 = 3;
+    const EINBALANCED_COIN_EXCHANGE: u64 = 4;
+
+    fun init_module(account: &signer) {
+        // get the resource signer
+        let resource_signer_cap = resource_account::retrieve_resource_account_cap(account, @0xcafe);
+        let resource_signer = account::create_signer_with_capability(&resource_signer_cap);
+
+        let (wrapped_burn_cap, freeze_cap, wrapped_mint_cap) = coin::initialize<WrappedAptosCoin>(
+            &resource_signer,
+            string::utf8(b"Wrapepd Aptos Coin"),
+            string::utf8(b"WAPT"),
+            8,
+            false,
+        );
+
+        coin::destroy_freeze_cap(freeze_cap);
+
+        coin::register<AptosCoin>(account);
+        coin::register<WrappedAptosCoin>(account);
+
+        move_to(account, ModuleData {
+            resource_signer_cap,
+        });
+
+        move_to(account, CoinCapabilities {
+            burn_cap: wrapped_burn_cap,
+            mint_cap: wrapped_mint_cap,
+        });
+    }
+
+    // Exchange AptosCoin to WrappedAptosCoin
```

Comment:
use /// for function comments so that the comment can be included auto-generated doc